src/jetstream/main.go

```diff
@@ -1112,11 +1118,20 @@ func (p *portalProxy) registerRoutes(e *echo.Echo, needSetupMiddleware bool) {
 
 	// API endpoints with Swagger documentation and accessible with an API key that require admin permissions
 	stableAdminAPIGroup := stableAPIGroup
-	stableAdminAPIGroup.Use(p.adminMiddleware)
+	if p.GetConfig().UserEndpointsEnabled != config.UserEndpointsConfigEnum.Disabled {
```

Comment:
While implementing this, i encountered a problem that reminded me of an issue with the e2e tests.

From slack:
>The issue revolved around the /api/v1/endpoints request and using it in a system with no endpoints. we only use the request in e2e tests, in the browser we fetch endpoints via a /info request.

When i created the new group, i couldn't create any endpoints because the middleware for checking EDIT/DELETE routes also applied to the CREATE route. The cause was this issue: https://github.com/labstack/echo/issues/1519

Contrary to the docs, the middleware was applied to the whole group, instead of just the routes after the `Echo#Use()` function. 

I don't think this had any effect on the e2e tests tho, since the middleware is only applied when `UserEndpointsEnabled` is set to something else than `disabled`, but i wanted to mention this anyways.

Reply:
I noticed that the e2e test fail, because they are receiving a html instead of json. My assumption is, that my `stableAdminAPIGroup.Group("/endpoints")` overwrites the path `stableAPIGroup.GET("/endpoints", p.listCNSIs)` and therefore the backend serves static files. I'm not sure i can solve this without changing paths or putting the GET route also in the group. I will look into it and give a report later if i found a solution.